### ISSUE TITLE
signout with RemoveAsync and signin with WithPrompt(Prompt.ForceLogin) does not ask password

### ISSUE THREAD

**Issue by viebrix** [AzureAD/microsoft-authentication-library-for-dotnet#4133]
**Which version of MSAL.NET are you using?**
MSAL.NET 4.53.0

**Platform**
.NET 4.8 (Winforms)

**What authentication flow has the issue?**
* Desktop / Mobile
    * [*] Interactive
    * [ ] Integrated Windows Authentication
    * [ ] Username Password
    * [ ] Device code flow (browserless)
* Web app 
    * [ ] Authorization code
    * [ ] On-Behalf-Of
* Daemon app 
    * [ ] Service to Service calls

Other?

**Is this a new or existing app?**
This is a new app

```csharp
//signout:
var accounts = await MyAuthentication.Pca.GetAccountsAsync();
if (accounts.Any())
{
		await MyAuthentication.Pca.RemoveAsync(accounts.FirstOrDefault());
		Logger.LogDebug($"User: {accounts.FirstOrDefault().Username} signed out!");
}

//signin:
authResult = await Pca.AcquireTokenInteractive(scopes).WithPrompt(Prompt.ForceLogin)
                                                  .ExecuteAsync()
                                                  .ConfigureAwait(false);

```

**Expected behavior**
I expected, that user is signed out and after requesting new token with AcquireTokenInteractive, the pop-up windows opens, where user can be chosen. After user (e-mail) is chosen, the user should again enter password.

**Actual behavior**
I expected, that user is signed out and after requesting new token with AcquireTokenInteractive, the pop-up windows opens, where user can be chosen. After user (e-mail) is chosen, the user is signed in without entering a password. 
Also after manual deleting cache file, user is automatically logged in.
This behavior is the same for two different logins, on same desktop pc, where the password was entered only one time.

**Comment by bgavrilMS** [comment]
This is by design @viebrix . The browser has cookies and also talks to Windows to bring SSO. The fact that the browser remembers the user, and that the user has signed into Windows with that account are user concerns. You, as an app owner, should not try to change the SSO behavior.

I don't recommend you use `Prompt.ForceLogin`, this creates a bad auth experience for customers. Why are trying to do this?

If you need to enforce a sing-in frequency, the tenant admin can do this via Conditional Access https://learn.microsoft.com/en-us/azure/active-directory/conditional-access/concept-conditional-access-session

**Comment by viebrix** [comment]
Thanks for the quick answer! I didn't use Prompt until I read on stack overflow and here on github, that this forces the user to reenter password.
My idea was, that my application uses msal to authenticate a user. But if for example a colleague (person b) signs in this app (on person a's computer) with his own credential (person b's credential) to show (person b) something. Person a is possible to relogin with person b's login without entering a password for a long time. This is an unwanted situation. I know person b works here under person a's desktop, but in practice this will happen, if both colleagues siting side by side to examine a task.

**Comment by bgavrilMS** [comment]
Have a look at https://learn.microsoft.com/en-us/azure/active-directory/conditional-access/howto-conditional-access-session-lifetime#persistence-of-browsing-sessions 

Otherwise, the SDK does not have control of the browser / can't clear cookies etc. For accounts connected to Windows, this should not be a concern. For accounts not connected to Windows, the user can explicitly sign-out, from the browser. You can see the 3 dots menu - click on that and say "Sign Out and Forget"

<img width="363" alt="image" src="https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/assets/12273384/dd4e14f2-da47-452a-a62d-ff0e8d71e413">

**Comment by viebrix** [comment]
Thanks I understand. I will read the docs you have provided. thanks very much for your help!

**Comment by viebrix** [comment]
@bgavrilMS sorry to ask you again. Maybe I missunderstod somehthing -  But I tried to remove the user in the browser(s) (with 3 dots) and the user is still in my winforms login msal window. In which browser will the cookie be stored for winforms msal interactive login?
login_firefox.jpg shows the https://login.microsoftonline.com site in my standard browser
login_chrome.jpg shows same address in browser i thought which will store cookies
login_msal_winforms.jpg shows the signin in my app.

![login_firefox](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/assets/33760394/a7211ef8-b101-4f9e-b917-e0be7e3b82bb)

![login_msal_winforms](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/assets/33760394/9064d32b-1ecc-4e86-9b9c-91180cca35f2)

![login_chrome](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/assets/33760394/1e3a2eb8-c46f-4205-8acb-3615e689a269)

**Comment by bgavrilMS** [comment]
Each browser is separate and cookies are not shared. The embedded browser is based on IE :( but it counts as a separate browser.

We now recommend folks to use WAM, see https://aka.ms/msal-net-wam as it provides better SSO
System browser is better than embedded because it supports FIDO / Windows Hello etc.

**Comment by viebrix** [comment]
I thought I'm already using WAM:
`   Pca = PublicClientApplicationBuilder.Create(ClientId)
              .WithAuthority(AuthorityUri)
              .WithLogging(Log, Microsoft.Identity.Client.LogLevel.Verbose, true)
              .WithRedirectUri("http://localhost") // required for DefaultOsBrowser
              .WithBroker(new BrokerOptions(BrokerOptions.OperatingSystems.Windows))
              .WithParentActivityOrWindow(WindowHandleProvider)
              //.WithWindowsDesktopFeatures(options);
              .Build();`

**Comment by bgavrilMS** [comment]
Ok got it. So are you saying that the WAM dialog does not display the 3 dots? Note that WAM is trying to get all new accounts in Windows. 

For Work and School accounts, the user is asked if the account should be "app only" or in Windows.
For Personal accounts, they do not get asked.

**Comment by viebrix** [comment]
Yes, it only shows the 3 dots in the right bottom corner, which only opens some status info about errors and so on. 
you can see the screenshot in my comment 1 hour ago. Its the one with "login_msal_winforms.jpg" - this is the WAM dialog.
Both accounts which are displayed are work accounts for (in?) my tenant. They are also windows accounts on AD and on my PC.
Sorry if I'm not clear in describing, but azure ad is new to me and so are all vocabulary.

**Comment by bgavrilMS** [comment]
Possibly. Today, MSAL does not pass on the Prompt option to WAM. That's a bug.

**Comment by viebrix** [comment]
Thanks for your help and analyzing the background of the issue!

**Comment by aries-zhang** [comment]
Sorry for the delay in raising a bug about the prompt to WAM. It seems this issue is now tracking it, so attaching the convo here: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4110#issuecomment-1530955885

**Comment by JHVIW** [comment]
Hi there! I'd love to know when this issue will be resolved. Unfortunately it was removed from release 4.59.0. 
My company is encountering the same problem and we'd love for it to be fixed so we can continue offering the best functionality to our customers.

Looking forward to your response!

**Comment by nicklasv** [comment]
Hello, do you have any information about this problem?

We have a need to enforce the password prompt to show every time the user login since the computer is shared between many persons. 

As far as I understand will a Conditional Access not be specific for our application but for all logins, browsers and other WPF clients, is this correct?

**Comment by JHVIW** [comment]
We haven’t been able to fully resolve this issue either, as the underlying bug is still present. However, we had a similar requirement and found a workaround that might help.

We configured our application to initiate a request to a specific page on our website. This page redirects the user to the Microsoft login screen using the Azure AD API, where we can enforce a sign-in every time. After the user completes the login, the page returns a response indicating whether the user successfully authenticated or not. Our application then reads this response to determine the login status and proceeds accordingly.
